URBNSwiftCarousel/Classes/URBNPresentationController.swift

```diff
@@ -0,0 +1,199 @@
+//
+//  URBNPresentationController.swift
+//  Pods
+//
+//  Created by Kevin Taniguchi on 4/30/16.
+//
+//
+
+import UIKit
+
+/**
+    This subclass of UIPresentationController manages the UIImageView that zooms in and out between the source and destination view controllers.
+    It also handles synchonizaton of the collection views
+*/
+
+public class URBNPresentationController: UIPresentationController {
+    
+    // MARK: Public Variables
+    public var maskingNavBarColor: UIColor?
+    
+    // MARK: Private Variables
+    private var transitionView = UIImageView()
+    
+    private let assertionWarning = "Warning : make sure  all VC's being passed in conform to the URBNSwCarouselTransitioning protocol"
+    
+    lazy private var navBarCoverView = UIView()
+    
+    private var topFromVC: URBNSwCarouselTransitioning? {
+        let nav = presentingViewController as? UINavigationController
+        return nav?.topViewController as? URBNSwCarouselTransitioning
+    }
+    
+    private var navBarOfPresentingViewController: UIView? {
+        guard let nav = presentingViewController as? UINavigationController else { return nil }
+        let snapShotOfNavbar = nav.navigationBar.snapshotViewAfterScreenUpdates(true)
+        snapShotOfNavbar.translatesAutoresizingMaskIntoConstraints = false
+        return snapShotOfNavbar
+    }
+    
+    // MARK: Public Overrides of UIPresentationController's view tracking methods
+    override public func presentationTransitionWillBegin() {
+        guard let destinationVC = presentedViewController as? URBNSwCarouselTransitioning, containingView = containerView, transitionCoordinator = presentingViewController.transitionCoordinator()
```

Comment:
ah well if not in the example, the conformance can be limited to here and there's no downside for the client. but there is a bunch of type checking in the example project that we should aim to clean up